CefSharp.BrowserSubprocess.Core/ConcurrentObjectRegistry.h

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright © 2010-2015 The CefSharp Project. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license that can be found in the LICENSE file.
+
+#pragma once
+
+#include "JavascriptCallbackWrapper.h"
+
+using namespace System::Collections::Concurrent;
+
+namespace CefSharp
+{
+    namespace Internals
+    {
+        generic <typename T>
+        public ref class ConcurrentObjectRegistry
```

Comment:
In general, I'm not convinced this class is actually required. It only removes a small amount of code duplication and introduces what I'm wondering is an unnecessary layer of abstraction.